src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs

```diff
@@ -34,13 +34,13 @@ public class SubscriptionsHandler : DatabaseRequestHandler
         }
 
         [RavenAction("/databases/*/subscriptions", "DELETE", "/databases/{databaseName:string}/subscriptions?id={subscriptionId:long}")]
-        public Task Delete()
+        public async Task Delete()
         {
             var id = GetLongQueryString("id").Value;
 
-            Database.SubscriptionStorage.DeleteSubscription(id);
+            await Database.SubscriptionStorage.DeleteSubscription(id);
```

Comment:
Yeah, that happens inside the SubscriptionStorage (like we've done with the indexes)